includes/API/ApiLabkiUpdate.php

```diff
@@ -11,165 +11,336 @@
+            $packIdObj = $packReg->registerPack( $repoId, $packName, $version, $this->getUser()->getId() );
+            $packId = $packIdObj->toInt();
 
-            case 'updatePack':
-                // Integration point: BEFORE updating registry, update MW pages (rename/move/edit)
-                // 1) Diff old vs new pages; create/rename/delete as needed in core
-                // 2) Only when those actions succeed, update registry below
-                // Update pack metadata (e.g., version) and replace its pages with provided list
-                $contentRepoUrl = (string)( $params['contentRepoUrl'] ?? '' );
-                $packName = (string)( $params['packName'] ?? '' );
-                $currentVersion = isset( $params['version'] ) ? (string)$params['version'] : null;
-                $newVersion = isset( $params['newVersion'] ) ? (string)$params['newVersion'] : null;
-                $pagesParam = $params['pages'] ?? null; // JSON or array
-                $pages = is_string( $pagesParam ) ? json_decode( $pagesParam, true ) : ( ( is_array( $pagesParam ) ) ? $pagesParam : [] );
-                if ( $contentRepoUrl === '' || $packName === '' ) {
-                    $this->dieWithError( 'apierror-missing-param' );
-                }
-                $repoId = $repoReg->ensureRepo( $contentRepoUrl );
-                $packIdObj = $packReg->getPackIdByName( $repoId, $packName, null );
-                if ( $packIdObj === null ) {
-                    $this->dieWithError( [ 'apierror-pack-not-found', $packName ], 'pack_not_found' );
-                }
-                $packId = $packIdObj instanceof PackId ? $packIdObj->toInt() : (int)$packIdObj;
-                if ( $newVersion !== null && $newVersion !== '' ) {
-                    $packReg->updatePack( $packId, [ 'version' => $newVersion ] );
-                } else {
-                    $packReg->updatePack( $packId, [] ); // touch updated_at
-                }
-                // Replace pages
-                $pageReg->removePagesByPack( $packId );
-                $added = 0;
-                foreach ( $pages as $p ) {
-                    if ( !is_array( $p ) ) { continue; }
-                    $name = (string)($p['name'] ?? '');
-                    $finalTitle = (string)($p['finalTitle'] ?? '');
-                    $ns = (int)($p['pageNamespace'] ?? 0);
-                    $wikiPageId = (int)($p['wikiPageId'] ?? 0);
-                    if ( $name === '' || $finalTitle === '' ) { continue; }
-                    $pageReg->recordInstalledPage( $packId, $name, $finalTitle, $ns, $wikiPageId );
-                    $added++;
-                }
-                $result['success'] = true;
-                $result['packId'] = $packId;
-                $result['pagesAdded'] = $added;
-                break;
+            $pageReg->removePagesByPack( $packId );
+            foreach ( $createdPages as $p ) {
+                $pageReg->recordInstalledPage(
```

Comment:
i see an `addPage` method and a `recordInstalledPage` method in the PageRegistry module, are they different?